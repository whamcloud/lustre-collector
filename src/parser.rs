// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

use crate::{mds::mds_parser, mgs::mgs_parser, oss::oss_parser, top_level_parser, types::Record};
use combine::{choice, error::ParseError, many, Parser, Stream};

pub fn params() -> Vec<String> {
    let mut a = top_level_parser::top_level_params();
    a.extend(mgs_parser::params());
    a.extend(oss_parser::params());
    a.extend(mds_parser::params());
    a
}

pub fn parse<I>() -> impl Parser<I, Output = Vec<Record>>
where
    I: Stream<Token = char>,
    I::Error: ParseError<I::Token, I::Range, I::Position>,
{
    many(choice((
        top_level_parser::parse(),
        mgs_parser::parse(),
        mds_parser::parse(),
        oss_parser::parse(),
    )))
}

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_debug_snapshot;

    #[test]
    fn test_params() {
        assert_debug_snapshot!(params());
    }
}
