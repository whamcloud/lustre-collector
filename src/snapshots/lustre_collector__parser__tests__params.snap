---
source: src/parser.rs
expression: params()
---
[
    "memused",
    "memused_max",
    "lnet_memused",
    "health_check",
    "mgs.*.mgs.stats",
    "mgs.*.mgs.threads_max",
    "mgs.*.mgs.threads_min",
    "mgs.*.mgs.threads_started",
    "mgs.*.num_exports",
    "obdfilter.*OST*.job_stats",
    "obdfilter.*OST*.stats",
    "obdfilter.*OST*.brw_stats",
    "obdfilter.*OST*.filesfree",
    "obdfilter.*OST*.filestotal",
    "obdfilter.*OST*.fstype",
    "obdfilter.*OST*.kbytesavail",
    "obdfilter.*OST*.kbytesfree",
    "obdfilter.*OST*.kbytestotal",
    "obdfilter.*OST*.num_exports",
    "obdfilter.*OST*.tot_dirty",
    "obdfilter.*OST*.tot_granted",
    "obdfilter.*OST*.tot_pending",
    "ldlm.namespaces.{mdt-,filter-}*.contended_locks",
    "ldlm.namespaces.{mdt-,filter-}*.contention_seconds",
    "ldlm.namespaces.{mdt-,filter-}*.ctime_age_limit",
    "ldlm.namespaces.{mdt-,filter-}*.early_lock_cancel",
    "ldlm.namespaces.{mdt-,filter-}*.lock_count",
    "ldlm.namespaces.{mdt-,filter-}*.lock_timeouts",
    "ldlm.namespaces.{mdt-,filter-}*.lock_unused_count",
    "ldlm.namespaces.{mdt-,filter-}*.lru_max_age",
    "ldlm.namespaces.{mdt-,filter-}*.lru_size",
    "ldlm.namespaces.{mdt-,filter-}*.max_nolock_bytes",
    "ldlm.namespaces.{mdt-,filter-}*.max_parallel_ast",
    "ldlm.namespaces.{mdt-,filter-}*.resource_count",
    "mdt.*.md_stats",
    "mdt.*MDT*.num_exports",
    "osd-*.*MDT*.filesfree",
    "osd-*.*MDT*.filestotal",
    "osd-*.*MDT*.kbytesavail",
    "osd-*.*MDT*.kbytesfree",
    "osd-*.*MDT*.kbytestotal",
]
